controllers/snapshot/snapshot_adapter.go

```diff
@@ -60,7 +63,41 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 		a.logger.Info("The Snapshot has finished testing.")
 		return results.ContinueProcessing()
 	}
+
 	a.logger.Info("Placeholder for triggering all Integration pipelines (including optional ones)")
+	integrationTestScenarios, err := helpers.GetAllIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+
+	if err != nil {
+		a.logger.Error(err, "Failed to get Integration test scenarios for following application",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+	}
+
+	for _, integrationTestScenarios := range *integrationTestScenarios {
```

Comment:
Maybe we can rename the `integrationTestScenarios` to `integrationTestScenario` singular for the loop variable as it is a single integrationTestScenario.